src/observability/assessment.go

```diff
@@ -0,0 +1,266 @@
+// Package scan to scan for risks
+package observability
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/accuknox/auto-policy-discovery/src/cluster"
+	opb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/observability"
+
+	//"github.com/rs/zerolog/log"
+	"golang.org/x/exp/slices"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+type Volmount struct {
+	Mounts    []string
+	Podname   string
+	MountType string
+}
+
+var po []Volmount
+
+func Scan(request *opb.Request) (*opb.AssessmentResponse, error) {
+
+	client := cluster.ConnectK8sClient()
+	var v *opb.AssessmentResponse
+	podList, err := client.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	fmt.Printf("There are %d Pods in the cluster\n", len(podList.Items))
+
+	//data := &opb.Request{
+	//	Label:         o.Labels,
+	//	NameSpace:     o.Namespace,
+	//	PodName:       o.PodName,
+	//	ClusterName:   o.ClusterName,
+	//	ContainerName: o.ContainerName,
+	//	Aggregate:     o.Aggregation,
+	//}
+	// create a client
+	var sumResponses []*opb.Response
+
+	if err != nil {
+		return nil, errors.New("could not connect to the server. Possible troubleshooting:\n- Check if discovery engine is running\n- kubectl get po -n accuknox-agents")
+	}
+
+	// create a client
+	//defer conn.Close()
+	//Sumclient := opb.NewObservabilityClient(conn)
+	//if err != nil {
+	//	fmt.Println(err)
+	//}
+	//podNameResp, err := GetPodNames(request)
+	//if err != nil {
+	//	fmt.Println(err)
+	//}
+	if len(podList.Items) > 0 {
+		//fmt.Println(pods.Name)
+		//containersSATokenMountPath, err := getSATokenMountPath(&pod)
+		if err != nil && strings.Contains(err.Error(), "service account token not mounted") {
+			log.Warn().Msg(err.Error())
+			return nil, err
+		}
+
+		//in := &opb.Request{
+		//	PodName:       pods.Name,
+		//	NameSpace:     pods.Namespace,
+		//	ContainerName: container.Name,
+		//	Type:          "process,file,network",
+		//}
+		sumResp, err := GetSummaryData(request)
+		if err != nil {
+			print("ERRRRRRRRRRRRRRRRRROR")
+			log.Warn().Msg(err.Error())
+			return nil, err
+		}
+		//log.Info().Msg(sumResp.String())
+		sumResponses = append(sumResponses, sumResp)
+
+		//fmt.Print(ShouldSATokenBeAutoMounted(), "\n\n\n")
+		v = VolumeUsed(sumResponses, podList)
+		//b, _ := json.MarshalIndent(test, "", "    ")
+		//fmt.Println(string(b))
```

Comment:
Please do not review on this file yet, full assessment logic has not been pushed yet, want reviews on my handling of protobuf handling first.